CMakeLists.txt

```
@@ -40,6 +40,61 @@ else()
   message(WARNING "Unknown compiler... recklessly proceeding without a version check")
 endif()
 
+set(LIBMONGOC_REQUIRED_VERSION 1.22.1) # TODO: update to 1.24.0 once released.
+set(LIBMONGOC_REQUIRED_ABI_VERSION 1.0)
+
+set(NEED_DOWNLOAD_C_DRIVER false CACHE INTERNAL "")
+if(TARGET mongoc_shared OR TARGET mongoc_static)
+  # If these targets exist, then libmongoc has already been included as a project
+  # sub-directory
+  message ("found libmongoc targets declared in current build scope; version not checked")
+
+  if(NOT MONGOCXX_LINK_WITH_STATIC_MONGOC)
+    set(libmongoc_target mongoc_shared)
+  else()
+    set(libmongoc_target mongoc_static)
+  endif()
+
+  if(MONGOCXX_BUILD_STATIC)
+    set(mongocxx_pkg_dep "find_dependency(mongoc-1.0 REQUIRED)")
+  endif()
+else()
+    find_package(mongoc-${LIBMONGOC_REQUIRED_ABI_VERSION} ${LIBMONGOC_REQUIRED_VERSION} QUIET)
+    if(mongoc-${LIBMONGOC_REQUIRED_ABI_VERSION}_FOUND)
+        message("already found libmongoc, don't need to download it.")
+    else()
+        set(NEED_DOWNLOAD_C_DRIVER true CACHE INTERNAL "")
+    endif()
+endif()
+
+if (${NEED_DOWNLOAD_C_DRIVER})
```

Boolean variables need not be dereferenced. This is one of the very weird quirks in CMake syntax: Refer to [Variable Expansion](https://cmake.org/cmake/help/latest/command/if.html#variable-expansion) on the page for `if()`.

```suggestion
if (NEED_DOWNLOAD_C_DRIVER)
```